polygon.js

```diff
@@ -0,0 +1,216 @@
+const PI                   = Math.PI,
+      NUMBER_OF_POLYGON    = 6,
+      MAX_NUMBER_OF_ANGLES = 7,
+      length               = 100;   
+
+function main() {
+    var canvas   = document.getElementById("canvas"),
+        ctx      = canvas.getContext('2d'),
+        polygons = [],
+        mouse = {
+            x: 0,
+            y: 0
+        }
+    canvas.setAttribute('height',window.innerHeight - 20);
+    canvas.setAttribute('width',window.innerWidth - 20);
+    polygons = initPolygon(NUMBER_OF_POLYGON, MAX_NUMBER_OF_ANGLES);
+    colorDraw(polygons, canvas, ctx);
+    canvas.addEventListener('mousedown', function(evt){
+        mouse.x = evt.clientX;
+        mouse.y = evt.clientY;
+        polygons.forEach(function(item, i, arr){
+            arr[i].selected = inPolygon(mouse.x, mouse.y, item);
+        });
+    });
+    canvas.addEventListener('mousemove', function(evt){
+        polygons.forEach(function(item, i, arr){
```

Comment:
How about create some service that will be responsible of saving selected item/items. Now this cycle can be called hundred times a second